File: solution/070ClimbingStairs/070ClimbingStairs.go
package main

func climbStairs(n int) int {
	if n < 2 {
		return 1
	}
	fn_2 := 1
	fn_1 := 1
	fn := 0
	for i := 2; i <= n; i++ {
		fn = fn_2 + fn_1
		fn_2 = fn_1
		fn_1 = fn
	}
	return fn
}

func main() {

}

File: solution/070ClimbingStairs/070ClimbingStairs.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        // f(n) = f(n-1)+f(n-2)
        if n == 1 {return 1;}
        let mut fn_2 = 1;
        let mut fn_1 = 1;
        let mut fn_0 = fn_1+fn_2;
        for i in 1..n {
            fn_0 = fn_1+fn_2;
            fn_2 = fn_1;
            fn_1 = fn_0;
        }
        fn_0
    }
}
